tests/test_service.py

```diff
@@ -153,6 +153,12 @@ def test_empty_data_invalid(self):
 
         self.assertInvalid(empty_data)
 
+    def test_census_empty_data_invalid(self):
+        empty_data = json.loads(self.message['0.0.2'])
+        empty_data['data'] = ""
```

Comment:
Could you do a test also with non-empty data that's just of the wrong format? Like:
```
data = { "key1": "Value1", "Key2": "Value2" } # key value pairs
data = ["a","b","c", {"Some": "Thing"}] # plain / non dict values in array
data = "abcdegh" # plain value
```
etc